### PR TITLE
Updated Color APIs

### PR DESCRIPTION
**Pull request by Licini**
1. More consistent way of dealing with colors:
Use `pointcolor`, `linecolor`, `facecolor` for single color for all elements
Use `pointcolors`, `linecolors`, `facecolors` for multiple colors per element.
See example at `scripts/v120_colors.py`
```
facecolors = {k: [random(), random(), random()] for k in mesh.faces()}
linecolors = {k: [random(), random(), random()] for k in mesh.edges()}
pointcolors = {k: [random(), random(), random()] for k in mesh.vertices()}

viewer.add(mesh, name="mesh1", show_points=True, facecolors=facecolors, linecolors=linecolors, pointcolors=pointcolors)
viewer.add(mesh2, name="mesh2", show_points=True, facecolor=[1, 0, 0], linecolor=[0, 1, 0], pointcolor=[0, 0, 1])

```
image

2. Visualisation tab added. Allowing editing in real time without updating the object data. 
image

3. Common Attributes like `xxxcolor` and `show_xxx` are now all concentrated to `BufferObject`, inheritance classes are simplified.

### CODE REVIEW

File: src/compas_view2/forms/editform.py [hunk 1]
```diff
@@ -168,46 +200,79 @@ def map_number(self, obj, attribute, name=None, layout=None, update_data=True):
 
         # layout.setContentsMargins(0, 0, 0, 0)
         label = QtWidgets.QLabel(name or str(attribute))
-        if isinstance(obj, list) or isinstance(obj, dict):
+        if type(obj) in [list, dict, np.ndarray]:
```
Comment (brgcode) [inline src/compas_view2/forms/editform.py:203]: why not `isinstance(obj, (list, dict, np.ndarray))`?

File: scripts/v120_colors.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
+from os import name
```
Comment (brgcode) [inline scripts/v120_colors.py:1]: seems accidental

File: src/compas_view2/objects/lineobject.py [hunk 3]
```diff
@@ -8,22 +8,20 @@ class LineObject(BufferObject):
     default_color_points = [0.1, 0.1, 0.1]
     default_color_lines = [0.4, 0.4, 0.4]
 
-    def __init__(self, data, show_lines=True, pointcolor=None, linecolor=None, color=None, **kwargs):
+    def __init__(self, data, show_lines=True, **kwargs):
         super().__init__(data, show_lines=show_lines, **kwargs)
```
Comment (brgcode) [inline src/compas_view2/objects/lineobject.py:12]: this seems a bit pointless

Comment (brgcode) [reply src/compas_view2/objects/lineobject.py:12]: or is it necessary for the name to be set properly?

File: src/compas_view2/objects/pointcloudobject.py [hunk 4]
```diff
@@ -28,14 +28,15 @@ class PointcloudObject(BufferObject):
         If number of colors does not equal to number of points.
     """
 
-    def __init__(self, data, color=None, colors=None, size=10, **kwargs):
+    def __init__(self, data, size=10, colors=None, **kwargs):
```
Comment (brgcode) [inline src/compas_view2/objects/pointcloudobject.py:31]: `pointsize` would make more sense in this case

**Review by brgcode — APPROVED** [review]
i think it would make sense to make the available initialisation options explicit in the `__init__` function of the individual objects, and document them there. currently users have to dig quite deep into the code to figure out which options are available...

File: src/compas_view2/forms/editform.py (hunk 1 above)
Comment (Licini) [reply src/compas_view2/forms/editform.py:203]: ha didn't know you can write like that!

File: scripts/v120_colors.py (hunk 2 above)
Comment (Licini) [reply scripts/v120_colors.py:1]: ..... damn, some auto imports

**Comment by Licini** [comment]
> i think it would make sense to make the available initialisation options explicit in the `__init__` function of the individual objects, and document them there. currently users have to dig quite deep into the code to figure out which options are available...

I have a bit of mixed feeling on this tho. Totally agree that it's very tricky from users perspective to know what are available options when looking at source code or autogen docs. But on the other side having universal attributes like `facecolor` duplicated in every children classes will increase a lot of maintenance efforts and create many possible places of error. It's almost a un-avoidable nature of Inheriting classes being double-edged sword.... Perhaps we can open Another PR specifically for possible solutions on this?

**Comment by Licini** [comment]
> > i think it would make sense to make the available initialisation options explicit in the `__init__` function of the individual objects, and document them there. currently users have to dig quite deep into the code to figure out which options are available...
> 
> I have a bit of mixed feeling on this tho. Totally agree that it's very tricky from users perspective to know what are available options when looking at source code or autogen docs. But on the other side having universal attributes like `facecolor` duplicated in every children classes will increase a lot of maintenance efforts and create many possible places of error. It's almost a un-avoidable nature of Inheriting classes being double-edged sword.... Perhaps we can open Another PR specifically for possible solutions on this?

Or perhaps, in documentation in each classes, we simply add section of inherited attributes??

**Comment by brgcode** [comment]
well i see it also as an opportunity to be a bit more specific about which parameters/attributes are relevant for each object because they are not all relevant all the time...

**Comment by brgcode** [comment]
but a separate PR for dealing with this sounds good

**Comment by Licini** [comment]
Ok mentioned issues fixed now merging

**Pull request merged**